presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -269,14 +277,40 @@ private PlanBuilder filter(PlanBuilder subPlan, Expression predicate, Node node)
         return subPlan.withNewRoot(new FilterNode(idAllocator.getNextId(), subPlan.getRoot(), rewrittenAfterSubqueries));
     }
 
-    private PlanBuilder project(PlanBuilder subPlan, Iterable<Expression> expressions)
+    private PlanBuilder project(PlanBuilder subPlan, Iterable<Expression> expressions, Node node)
     {
         TranslationMap outputTranslations = new TranslationMap(subPlan.getRelationPlan(), analysis);
 
         ImmutableMap.Builder<Symbol, Expression> projections = ImmutableMap.builder();
         for (Expression expression : expressions) {
             Symbol symbol = symbolAllocator.newSymbol(expression, analysis.getTypeWithCoercions(expression));
 
+            if (expression instanceof GroupingOperation && node instanceof QuerySpecification) {
+                FunctionCall rewrittenExpression = subPlan.rewriteGroupingOperationToFunctionCall((GroupingOperation) expression, (QuerySpecification) node);
```

Comment:
If it's going to stay like this, it should be factored out. I don't think it will though, this spraying of special cases all over the place is pretty hacky.